src/sonic_ax_impl/mibs/ietf/rfc4363.py

```diff
@@ -27,7 +20,19 @@ def __init__(self):
         self.vlanmac_ifindex_map = {}
         self.vlanmac_ifindex_list = []
         self.if_bpid_map = {}
-
+        self.bvid_vlan_map = {}
+
+    def fdb_vlanmac(self, fdb):
+        if 'vlan' in fdb:
+            vlan_id = fdb["vlan"]
+        elif 'bvid' in fdb:
+            if fdb["bvid"] in self.bvid_vlan_map:
+                vlan_id = self.bvid_vlan_map[fdb["bvid"]]
+            else:
+                vlan_id = port_util.get_vlan_id_from_bvid(self.db_conn, fdb["bvid"])
```

Comment:
>vlan_id [](http://example.com/codeflow?start=16&length=7)

We found an extreme case that for whatever reason, `get_vlan_id_from_bvid` returns `None` once, and it will stored into `bvid_vlan_map` as cache. It is recoverable, since `reinit_data` will clear `bvid_vlan_map` regularly. However, this should not be cached, so the recovery will be quick.

Reply:
In which SONiC branch did you see this issue?  From the code, it is possible to get `None` from `get_vlan_id_from_bvid`, and you should see the error log?   and I agree if for some reason we got `None`, it should not be cached, that seemed to be a straightforward change in the code with a simple check before saving to cache.

Reply:
202012. Could you provide a fix and possibly a testcase?